lib/mastermind.rb

```diff
@@ -0,0 +1,54 @@
+class Mastermind
+  attr_reader :passcode, :input, :guess_attempts, :output
+  def initialize(passcode:, input: $stdin, output: $stdout, guess_attempts: 4)
+    @passcode = passcode
+    # @correct_answers = 0
+    @input = input
+    @guess_attempts = 5
+    @output = output
+  end
+
+  def mastermind
+    guess = nil
+    current_attempt = 1
+    correct = 0
+    while correct < guess_attempts
+      guess = input.split("\n")
+
+      if guess.length != 4
+        raise ArgumentError, "Number of guess not completed, try again!!"
+      end
+
+      guess1 = guess[0]
```

Comment:
Try to extract the concept of what the number 4 means here, so you wouldn't have to enumerate each guess with a suffix of a number. Any chance it's `@guess_attempts`?

Reply:
Yes, chances is what I meant by guess_attempts